vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_recipient_id/post_recipient_id_gemini.cc

```diff
@@ -0,0 +1,113 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "bat/ledger/internal/endpoint/gemini/post_recipient_id/post_recipient_id_gemini.h"
+
+#include <utility>
+
+#include "base/base64.h"
+#include "base/guid.h"
+#include "base/json/json_reader.h"
+#include "base/json/json_writer.h"
+#include "bat/ledger/internal/endpoint/gemini/gemini_utils.h"
+#include "bat/ledger/internal/ledger_impl.h"
+#include "third_party/abseil-cpp/absl/types/optional.h"
+
+using std::placeholders::_1;
+
+namespace ledger {
+namespace endpoint {
+namespace gemini {
+
+PostRecipientId::PostRecipientId(LedgerImpl* ledger) : ledger_(ledger) {
+  DCHECK(ledger_);
+}
+
+PostRecipientId::~PostRecipientId() = default;
+
+std::string PostRecipientId::GetUrl() {
+  return GetApiServerUrl("/v1/payments/recipientIds");
+}
+
+type::Result PostRecipientId::ParseBody(const std::string& body,
+                                        std::string* recipient_id) {
+  DCHECK(recipient_id);
+
+  absl::optional<base::Value> value = base::JSONReader::Read(body);
+  if (!value || !value->is_dict()) {
+    BLOG(0, "Invalid JSON");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  base::DictionaryValue* dictionary = nullptr;
+  if (!value->GetAsDictionary(&dictionary)) {
+    BLOG(0, "Invalid JSON");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  const auto* result = dictionary->FindStringKey("result");
+  if (!result || *result != "OK") {
+    BLOG(0, "Failed creating recipient_id");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  const auto* id = dictionary->FindStringKey("recipient_id");
+  if (!id) {
+    BLOG(0, "Response missing a recipient_id");
+    return type::Result::LEDGER_ERROR;
+  }
+
+  *recipient_id = *id;
+  return type::Result::LEDGER_OK;
+}
+
+std::string PostRecipientId::GeneratePayload() {
+  base::Value payload(base::Value::Type::DICTIONARY);
+  payload.SetStringKey("label", base::GenerateGUID());
+
+  std::string json;
+  base::JSONWriter::Write(payload, &json);
+
+  std::string base64;
+  base::Base64Encode(base::StringPiece(json), &base64);
```

Comment:
I think `std::string` will implicitly convert to `StringPiece`; if so, the explicit `StringPiece` constructor call isn't necessary.

Reply:
resolved